packages/components/src/internal/components/search/SampleFinderSection.tsx

```diff
@@ -0,0 +1,183 @@
+import { Section } from '../base/Section';
+import { EntityDataType } from '../entities/models';
+import { capitalizeFirstChar } from '../../util/utils';
+import { EntityFieldFilterModal } from './EntityFieldFilterModal';
+import { AuditBehaviorTypes, Filter } from '@labkey/api';
+import { SchemaQuery } from '../../../public/SchemaQuery';
+import { FilterCardProps, FilterCards } from './FilterCards';
+import { SamplesTabbedGridPanel } from '../samples/SamplesTabbedGridPanel';
+import { SAMPLE_DATA_EXPORT_CONFIG, SAMPLE_STATUS_REQUIRED_COLUMNS } from '../samples/constants';
+import { SCHEMAS } from '../../schemas';
+import { InjectedQueryModels, withQueryModels } from '../../../public/QueryModel/withQueryModels';
+import { User } from '../base/models/User';
+import { SamplesEditableGridProps } from '../samples/SamplesEditableGrid';
+import { getFinderStartText } from './utils';
+
+const SAMPLE_FINDER_TITLE = "Find Samples";
+const SAMPLE_FINDER_CAPTION = "Find samples that meet all of these criteria";
+
+interface SampleFinderSamplesGridProps {
+    user: User;
+    getSampleAuditBehaviorType: () => AuditBehaviorTypes;
+    samplesEditableGridProps: Partial<SamplesEditableGridProps>;
+}
+
+interface Props extends SampleFinderSamplesGridProps {
+    parentEntityDataTypes: EntityDataType[];
+}
+
+interface SampleFinderHeaderProps {
+    parentEntityDataTypes: EntityDataType[],
+    onAddEntity: (entityType: EntityDataType) => void;
+}
+
+export const SampleFinderHeaderButtons: FC<SampleFinderHeaderProps> = memo(props => {
+    const { parentEntityDataTypes, onAddEntity } = props;
+
+    return (
+        <div>
+            Search by:
+            {parentEntityDataTypes.map((parentEntityType) => (
+                <button key={parentEntityType.nounSingular} className="btn btn-default margin-left" onClick={() => { onAddEntity(parentEntityType) }}>
+                    <i className="fa fa-plus-circle container--addition-icon" /> {capitalizeFirstChar(parentEntityType.nounAsParentSingular)} Properties
+                </button>
+            ))}
+        </div>
+    )
+});
+
+export const SampleFinderSection: FC<Props> = memo((props) => {
+    const { parentEntityDataTypes, ...gridProps } = props;
+
+    const [showFilterModal, setShowFilterModal] = useState<boolean>(false);
+    const [chosenEntityType, setChosenEntityType] = useState<EntityDataType>(undefined);
+    const [filterCards, setFilterCards] = useState<FilterCardProps[]>([]);
+
+    const onAddEntity = useCallback((entityType: EntityDataType) => {
+        setChosenEntityType(entityType);
+        setShowFilterModal(true);
+    }, []);
+
+    const onFilterEdit = useCallback((index: number) => {
+        console.log("onFilterEdit for index " + index + ": Not yet implemented.");
+        setShowFilterModal(true);
+    }, [filterCards]);
+
+    const onFilterDelete = useCallback((index: number) => {
+        const newFilterCards = [...filterCards];
+        newFilterCards.splice(index, 1);
```

Comment:
I'll note this in my manual testing, but when I add 3 or 4 filter cards for a given type and then click to delete the 2nd one. It removes all of the cards after that index (i.e. if I have 4 and click delete for the second, I end up with just the first card). This code seems correct, so not sure where that behavior is coming from.